packages/components/src/presentation-components/tree/controlled/PresentationTreeRenderer.tsx

```diff
@@ -36,23 +47,26 @@ export interface PresentationTreeRendererProps extends TreeRendererProps {
  */
 export function PresentationTreeRenderer(props: PresentationTreeRendererProps) {
   const nodeLoader = props.nodeLoader;
-
   const { applyFilter, clearFilter } = useHierarchyLevelFiltering({ nodeLoader, modelSource: nodeLoader.modelSource });
   const [filterNode, setFilterNode] = useState<PresentationTreeNodeItem>();
+  const treeModel = useMemo(() => props.visibleNodes.getModel(), [props.visibleNodes]);
 
   const filterableNodeRenderer = useCallback(
     (nodeProps: TreeNodeRendererProps) => {
       return (
         <PresentationTreeNodeRenderer
           {...nodeProps}
-          onFilterClick={(node) => {
-            setFilterNode(node);
+          onFilterClick={(nodeId) => {
+            const node = treeModel.getNode(nodeId);
```

Comment:
Better use `nodeLoader.modelSource.getModel()`. That way you don't need `useMemo`